client/settings/general-settings/test/general-settings.test.js

```diff
@@ -71,4 +71,43 @@ describe( 'GeneralSettings', () => {
 			);
 		}
 	);
+
+	it.each( [ [ true ], [ false ] ] )(
+		'displays Test Mode enabled = %s state from data store',
+		( isEnabled ) => {
+			useTestMode.mockReturnValue( [ isEnabled, jest.fn() ] );
+			render( <GeneralSettings /> );
+			const enableTestModeCheckbox = screen.getByLabelText(
+				'Enable test mode'
+			);
+
+			let expectation = expect( enableTestModeCheckbox );
+			if ( ! isEnabled ) {
+				expectation = expectation.not;
+			}
+			expectation.toBeChecked();
+		}
+	);
+
+	it.each( [ [ true ], [ false ] ] )(
+		'Confirmation Modal display when Test Mode = %s',
```

Comment:
```suggestion
		'Checks Dev Modal should display when initial testMode = %s',
```

Reply:
> 'Checks Dev Modal should display when initial testMode = %s',

Iis a test case which runs on both true and false, so it should be 
"Dev Modal should display when initial testMode = false"
"Dev Modal should **not** display when initial testMode = true"
which is a bit challenging to write without some complex formatting, I think?

Reply:
That looks good to me but I would clarify which modal because I think there is at least another confirmation modal (manual capture) in this settings page.

Reply:
Done